src/tools/ceph_dedup/ceph_dedup_daemon.cc

```diff
@@ -104,7 +107,7 @@ class SampleDedupWorkerThread : public Thread
     }
 
     /// increments refcount for entry, promotes as necessary, entry must be valid
-    entry_t increment_reference(entry_t entry) {
+    entry_t increment_reference(entry_t& entry) {
```

Comment:
I'm not sure this is desired.  This method is passed one entry and can potentially produce another one, so a signature that suggests that the entry is modified in-place is a bit misleading.  In addition `entry_t` is a tiny object (a enum constant + an iterator) and is routinely copied elsewhere in the code.

Reply:
makes sense! dropped this change in the latest revision.